### DISCUSSION TITLE
Could we have a single window mode, please?

### DISCUSSION THREAD

**Discussion by ckilb** [spatie/ray#730] [Ideas]
I really like Ray and even if I got to know it just a few days ago it's now part of my daily routine to use it.

One thing I'm really missing is a single window mode. For now Ray will create a new window once in a while.
I didn't figure out when exactly Ray is doing that. For sure this is the case if I debug between different projects (Docker vs SSH) - but sometimes even in between  for the same project. Maybe if I restart the Docker container.

As there is no project identifier (a label or something) in a Ray window I sometimes mistake a window with another (maybe newly opened) one... "Huh, no debug logs? Ah wrong window.".
I think usage would be much more conventient if there's a single window mode - so I can see all open windows at once as tabs. And it would be nice if the tabs have some meaningful name (like the directory  the debugging request came from).

_Actually, there's a Mac setting ("System Preferences" >"General" > "Prefer tabs") which is ignored by Ray though. I know about it because there was a similar discussion with Intellij Idea -[they fixed that once but now the bug is existing again](https://youtrack.jetbrains.com/issue/JBR-2893)._

**Comment by freekmurze** [comment]
You can solve this by setting the "Open new window" setting to "Never" in Ray's preferences.

![CleanShot 2022-09-14 at 18 02 48@2x](https://user-images.githubusercontent.com/483853/190205346-c15e2e31-c68b-4b8c-8f37-f1219cd7411b.jpg)
